app/core/middleware.py

```
@@ -163,3 +181,38 @@ def get_real_ip(request: Request) -> str:
+    async def dispatch(self, request: Request, call_next) -> Response:
+        if not ENABLE_METRICS or request.url.path == "/metrics":
+            return await call_next(request)
+            
+        start_time = time.perf_counter()
+        
+        try:
+            response = await call_next(request)
+            status_code = response.status_code
+        except Exception as e:
+            status_code = 500
+            raise e
```

Ruff (0.15.2)

[warning] Use `raise` without specifying exception name

Remove exception name

(TRY201)